### PR TITLE
GUARD-2572 Orders Sync Avatax Issue

### PR DESCRIPTION
**Pull request by maxim-skuvault**
[GUARD-2572]

- Change the legacy Order model's tax_lines.rate_id to string. WooCommerce with AVATAX plugin will return non-numeric text there instead of `int` (per API spec)
  - Tests for this use case will be added in [wooCommerceAccess](https://github.com/skuvault-integrations/wooCommerceAccess), to have all access tests in one place and also because adding tests in this repo would require moving a lot of files around
- On deserialization error, add the raw JSON response to the exception message and rethrow, to log more useful exceptions. We get a lot of WooCommerce deserialization exceptions
- Add the standard SkuVault pull request template

[GUARD-2572]: https://agileharbor.atlassian.net/browse/GUARD-2572?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by christian-bejarano-skuvault — APPROVED** [review]

**Review by christian-bejarano-skuvault — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Description
+
+Tickets: [JIRA-1234] (Replace with appropriate tickets. Some automations depend on this section, don't remove)
```
Comment (chris-skuvault) [inline .github/pull_request_template.md:3]: This has been updated in v1. Take a look https://github.com/skuvault/skuvault-v1/blob/staging/.github/pull_request_template.md  and make appropriate changes.

File: RestAPI.cs [hunk 2]
```diff
@@ -345,9 +345,18 @@ public virtual T DeserializeJSon<T>(string jsonString)
             };
 
             DataContractJsonSerializer ser = new DataContractJsonSerializer(typeof(T), settings);
-            MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(jsonString));
-            T obj = (T)ser.ReadObject(stream);
-            stream.Dispose();
+            T obj;
+            try
+            {
+                MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(jsonString));
```
Comment (chris-skuvault) [inline RestAPI.cs:351]: Why not use a `using` here?

File: .github/pull_request_template.md (hunk 1 above)
Comment (maxim-skuvault) [reply .github/pull_request_template.md]: Done

File: RestAPI.cs (hunk 2 above)
Comment (maxim-skuvault) [reply RestAPI.cs]: Makes sense. There's an explicit `stream.Dispose();` [right after](https://github.com/skuvault-integrations/WooCommerce.NET/pull/18/files#diff-d0540787297f5c9642e701956133276eba7c08f9f189e6f872d45b7b1c7fbf17R353). But if it fails to deserialize and throws then it probably won't fire this `Dispose`, unless I added it to `finally` as well. Added `using`.

Comment (chris-skuvault) [reply RestAPI.cs]: Using will dispose inside a finally on its own https://stackoverflow.com/questions/212198/what-is-the-c-sharp-using-block-and-why-should-i-use-it

**Review by chris-skuvault — APPROVED** [review]

**Pull request merged**